src/data/qpf/univariate/basic.lean

```diff
@@ -634,4 +642,32 @@ begin
   rw [supp_eq_of_is_uniform h, supp_eq_of_is_uniform h, image_comp]
 end
 
+theorem supp_preservation_iff_uniform :
+  q.supp_preservation ↔ q.is_uniform :=
+begin
+  split,
+  { intros h α a a' f f' h',
+    rw [← pfunctor.supp_eq,← pfunctor.supp_eq,← h,h',h] },
+  { rintros h α ⟨a,f⟩, rwa [supp_eq_of_is_uniform,pfunctor.supp_eq], }
+end
+
+theorem supp_preservation_iff_liftp_preservation :
+  q.supp_preservation ↔ q.liftp_preservation :=
+begin
+  split; intro h,
+  { rintros α p ⟨a,f⟩,
+    have h' := h, rw supp_preservation_iff_uniform at h',
+    dsimp [supp_preservation,supp] at h,
+    rwa [liftp_iff_of_is_uniform,supp_eq_of_is_uniform,pfunctor.liftp_iff'];
+      try { assumption },
+    { simp, split; intros; subst_vars; solve_by_elim } },
+  { rintros α ⟨a,f⟩,
+    simp [liftp_preservation] at h,
```

Comment:
Can you replace the nonterminal simps here?

Reply:
I squeezed most of them. One `dsimp` resisted squeezing, I'm not sure why but it looks fairly harmless

Reply:
Thanks! I'm not so worried about `dsimp`, it's much less brittle than `simp`.